File: 796. Rotate String/solution.cpp
class Solution {
public:
  bool rotateString(string s, string goal) {
    if (s.size() != goal.size()){
      return false;
    }

    for (int index = 0; index < s.size(); index++){
      if (s == goal){
        return true;
      }
      s = s.substr(1, s.size()) + s[0];
    }

    return false;
  }
};

File: 796. Rotate String/solution.js
/**
 * @param {string} s
 * @param {string} goal
 * @return {boolean}
 */
 var rotateString = function(s, goal) {
  if (s.length != goal.length){
    return false;
  }

  for (let index = 0; index < s.length; index++){
    if (s == goal){
      return true;
    }
    s = s.slice(1, s.length) + s[0];
  }

  return false;
};